instrumentation/github.com/gocql/gocql/config.go

```diff
@@ -0,0 +1,129 @@
+func (o TracedSessionOptionFunc) Apply(c *TracedSessionConfig) {
+	o(c)
+}
+
+// ------------------------------------------ TracedSessionOptions
+
+// WithQueryObserver sets an additional QueryObserver to the session configuration. Use this if
+// there is an existing QueryObserver that you would like called. It will be called after the
+// OpenTelemetry implementation, if it is not nil. Defaults to nil.
+func WithQueryObserver(observer gocql.QueryObserver) TracedSessionOption {
+	return TracedSessionOptionFunc(func(cfg *TracedSessionConfig) {
+		cfg.queryObserver = observer
+	})
+}
+
+// WithBatchObserver sets an additional BatchObserver to the session configuration. Use this if
+// there is an existing BatchObserver that you would like called. It will be called after the
+// OpenTelemetry implementation, if it is not nil. Defaults to nil.
+func WithBatchObserver(observer gocql.BatchObserver) TracedSessionOption {
+	return TracedSessionOptionFunc(func(cfg *TracedSessionConfig) {
+		cfg.batchObserver = observer
+	})
+}
+
+// WithConnectObserver sets an additional ConnectObserver to the session configuration. Use this if
+// there is an existing ConnectObserver that you would like called. It will be called after the
+// OpenTelemetry implementation, if it is not nil. Defaults to nil.
+func WithConnectObserver(observer gocql.ConnectObserver) TracedSessionOption {
+	return TracedSessionOptionFunc(func(cfg *TracedSessionConfig) {
+		cfg.connectObserver = observer
+	})
+}
+
+// WithTracer will set tracer to be the tracer used to create spans
+// for query, batch query, and connection instrumentation.
+// Defaults to global.Tracer("go.opentelemetry.io/contrib/github.com/gocql/gocql").
+func WithTracer(tracer trace.Tracer) TracedSessionOption {
+	return TracedSessionOptionFunc(func(c *TracedSessionConfig) {
+		c.tracer = tracer
+	})
+}
+
+// WithQueryInstrumentation will enable and disable instrumentation of
+// queries. Defaults to enabled.
+func WithQueryInstrumentation(enabled bool) TracedSessionOption {
+	return TracedSessionOptionFunc(func(cfg *TracedSessionConfig) {
+		cfg.instrumentQuery = enabled
+	})
+}
+
+// WithBatchInstrumentation will enable and disable insturmentation of
+// batch queries. Defaults to enabled.
+func WithBatchInstrumentation(enabled bool) TracedSessionOption {
+	return TracedSessionOptionFunc(func(cfg *TracedSessionConfig) {
+		cfg.instrumentBatch = enabled
+	})
+}
+
+// WithConnectInstrumentation will enable and disable instrumentation of
+// connection attempts. Defaults to enabled.
+func WithConnectInstrumentation(enabled bool) TracedSessionOption {
+	return TracedSessionOptionFunc(func(cfg *TracedSessionConfig) {
+		cfg.instrumentConnect = enabled
+	})
+}
+
+// ------------------------------------------ Private Functions
+
+func configure(options ...TracedSessionOption) *TracedSessionConfig {
+	config := &TracedSessionConfig{
+		tracer:            global.Tracer("go.opentelemetry.io/contrib/github.com/gocql/gocql"),
```

Comment:
Might be worth making this into an unexported constant to share across the package (i.e. the `Meter` as well).

```suggestion
		tracer:            global.Tracer("go.opentelemetry.io/contrib/instrumentation/github.com/gocql/gocql"),
```